### PR TITLE
Add debug logs to P2P scheduler plugin

### PR DESCRIPTION
**Pull request by hendrikmakait**
Adds debug logs to actions taken by the scheduler login

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: distributed/shuffle/_scheduler_plugin.py [hunk 1]
```diff
@@ -256,6 +261,7 @@ def _restart_shuffle(
         recs = self._restart_recommendations(id)
         self.scheduler.transitions(recs, stimulus_id=stimulus_id)
         self.scheduler.stimulus_queue_slots_maybe_opened(stimulus_id=stimulus_id)
+        logger.debug("Shuffle %s restarted due to stimulus '%s", id, stimulus_id)
```
[the remark this answers is not in the document]
Comment (hendrikmakait) [reply distributed/shuffle/_scheduler_plugin.py:264]: Works for me, these were primarily introduced for helping to debug Heisenberg bugs that's why I defaulted to `debug`. (Also P2P already produces a lot of spammy logs.)

[the remark this answers is not in the document]
Comment (hendrikmakait) [reply distributed/shuffle/_scheduler_plugin.py]: Done.

File: distributed/shuffle/_scheduler_plugin.py [hunk 2]
```diff
@@ -81,6 +82,10 @@ async def barrier(self, id: ShuffleId, run_id: int, consistent: bool) -> None:
         if shuffle.run_id != run_id:
             raise ValueError(f"{run_id=} does not match {shuffle}")
         if not consistent:
+            logger.debug(
```
[the remark this answers is not in the document]
Comment (hendrikmakait) [reply distributed/shuffle/_scheduler_plugin.py]: Done.

**Review by fjetter — APPROVED** [review]
I think we'll notice if this becomes too invasive and we can adjust. Thanks

**Pull request merged**